File: core/business-tier/src/main/scala/ru/korus/tmis/laboratory/across/accept/AntibioticSensitivity.scala
package ru.korus.tmis.laboratory.across.accept

import ru.korus.tmis.laboratory.across.{accept2 => lab2}
import scala.language.implicitConversions

final object AntibioticSensitivity {

  implicit def fromLab2(v: lab2.AntibioticSensitivity) = {
    AntibioticSensitivity(Option(v.getCode), Option(v.getName), Option(v.getValue), Option(v.getMic))
  }
}

/**
 * Чувствительность микроорганизма к антибиотикам
 */
final case class AntibioticSensitivity(
                                        /**
                                         * Идентификатор антибиотика по БД ЛИС
                                         */
                                        code: Option[String],

                                        /**
                                         * Название антибиотика
                                         */
                                        name: Option[String],

                                        /**
                                         * Чувствительность
                                         */
                                        value: Option[String],

                                        /**
                                         * Количественная характеристика чувствительности, если используется анализатор
                                         */
                                        mic: Option[String]
                                        )

File: core/common/src/main/java/ru/korus/tmis/lis/data/AntibioticSensitivity.java
package ru.korus.tmis.lis.data;

import javax.xml.bind.annotation.XmlElement;

/**
 * Чувствительность микроорганизма к антибиотикам
 */
public class AntibioticSensitivity {

    /**
     * Идентификатор антибиотика по БД ЛИС
     */
    private String code;

    /**
     * Название антибиотика
     */
    private String name;

    /**
     * Чувствительность
     */
    private String value;

    /**
     * Количественная характеристика чувствительности, если используется анализатор
     */
    private String mic;

    @XmlElement(name = "antibioticLisId")
    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    @XmlElement(name = "antibioticName", required = true)
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @XmlElement(name = "antibioticActivityValue", required = true)
    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    @XmlElement(name = "MIC")
    public String getMic() {
        return mic;
    }

    public void setMic(String mic) {
        this.mic = mic;
    }
}
